### ISSUE TITLE
 The input of 'fc' layer must be matrix 

### ISSUE THREAD

**Issue by OleNet** [PaddlePaddle/Paddle#708]
I1202 18:25:21.050037 6214 Util.cpp:113] Calling runInitFunctions 
I1202 18:25:21.050499 6214 Util.cpp:126] Call runInitFunctions done. 
I1202 18:25:21.261510 6214 Trainer.cpp:169] trainer mode: Normal 
I1202 18:25:21.261693 6214 MultiGradientMachine.cpp:108] numLogicalDevices=1 numThreads=4 numDevices=4 
I1202 18:25:21.277547 6214 PyDataProvider2.cpp:219] loading dataprovider dataprovider::process 
I1202 18:25:21.285418 6214 PyDataProvider2.cpp:219] loading dataprovider dataprovider::process 
I1202 18:25:21.285621 6214 GradientMachine.cpp:134] Initing parameters.. 
I1202 18:25:21.289291 6214 GradientMachine.cpp:141] Init parameters done. 
F1202 18:25:21.826174 6416 FullyConnectedLayer.cpp:86] Check failed: input.value The input of 'fc' layer must be matrix 
F1202 18:25:21.830188 6428 FullyConnectedLayer.cpp:86] Check failed: input.value The input of 'fc' layer must be matrix 
F1202 18:25:21.830188 6428 FullyConnectedLayer.cpp:86] Check failed: input.value The input of 'fc' layer must be matrixF1202 18:25:21.830322 6424 FullyConnectedLayer.cpp:86] Check failed: input.value The input of 'fc' layer must be matrix 
F1202 18:25:21.830188 6428 FullyConnectedLayer.cpp:86] Check failed: input.value The input of 'fc' layer must be matrixF1202 18:25:21.830322 6424 FullyConnectedLayer.cpp:86] Check failed: input.value The input of 'fc' layer must be matrixF1202 18:25:21.832275 6420 FullyConnectedLayer.cpp:86] Check failed: input.value The input of 'fc' layer must be matrix

**Comment by OleNet** [comment]
def lstm_dnn_net(input_dim,
                 class_dim,
                 emb_dim=128,
                 lstm_dim=256,
                 is_predict=False):
    """
    logistic
    """
    #bias_attr = ParameterAttribute(initial_std=0., l2_rate=0.0001)
    #lstm_para_attr = ParameterAttribute(initial_std=0., learning_rate=2e-3)  # , sparse_update=True)
    #para_attr = lstm_para_attr

    bias_attr = ParamAttr(initial_std=0.,l2_rate=0.)
    word_data = data_layer("word", input_dim)

    output = fc_layer(input=word_data,
                      size=class_dim,
                      act=SoftmaxActivation())

    if is_predict:
        outputs(output)
    else:
        outputs(
            classification_cost(input=output, label=data_layer('label', 1),
                                evaluator=[precision_recall_evaluator, classification_error_evaluator]))

**Comment by OleNet** [comment]
```
def lstm_dnn_net(input_dim,
                 class_dim,
                 emb_dim=128,
                 lstm_dim=256,
                 is_predict=False):
    """
    logistic
    """
    #bias_attr = ParameterAttribute(initial_std=0., l2_rate=0.0001)
    #lstm_para_attr = ParameterAttribute(initial_std=0., learning_rate=2e-3)  # , sparse_update=True)
    #para_attr = lstm_para_attr

    bias_attr = ParamAttr(initial_std=0.,l2_rate=0.)
    word_data = data_layer("word", input_dim)

    output = fc_layer(input=word_data,
                      size=class_dim,
                      act=SoftmaxActivation())

    if is_predict:
        outputs(output)
    else:
        outputs(
            classification_cost(input=output, label=data_layer('label', 1),
                                evaluator=[precision_recall_evaluator, classification_error_evaluator]))
```

**Comment by OleNet** [comment]
想实现一个logistic网络，但是却报：The input of 'fc' layer must be matrix。
网络如上。 是哪里写错了吗

**Comment by OleNet** [comment]
```
def hook(obj, dictionary, **kwargs):
    obj.input_types = [
        #integer_value_sequence(len(dictionary)),
        sparse_non_value_slot(len(dictionary)),
        integer_value(2)]
    obj.dictionary = dictionary
    obj.logger.info('dict len : %d' % (len(obj.dictionary)))
```
把值设置成sparse_non_value_slot才可以
